### ISSUE TITLE
Convert APIs using File to Path

### ISSUE THREAD

**Issue by jeenbroekstra** [eclipse/rdf4j#456]
(Migrated from https://openrdf.atlassian.net/browse/SES-1791)

As part of the Java-7 changes, we can support asynchronous IO calls and other enhancements by converting from File to Path in API calls.

http://docs.oracle.com/javase/tutorial/essential/io/legacy.html

**Comment by barthanssens** [comment]
According to the same log, this is related to #476

**Comment by barthanssens** [comment]
And replace some try-finally by try-with-resources in org.eclipse.rdf4j.common.io etc

**Comment by jeenbroekstra** [comment]
@barthanssens be aware that any changes in our public APIs for this will have to happen as part of a _major_ new release - unless you can implement in a way that is backward compatible.

**Comment by barthanssens** [comment]
Yeah, I was thinking on changing some internal stuff first, without changing the signatures of the methods... (breaking changes indeed require a major release)

**Comment by jeenbroekstra** [comment]
@barthanssens is this issue still on your radar?
